### DISCUSSION TITLE
Links

### DISCUSSION THREAD

**Discussion by erlend-sh** [muni-town/weird#64] [Ideas]
We want this in Weird:

https://blog.erlend.sh/evergreen-content-gardens

`links` is effectively a 'Weird primitive'. linktree style link additions for profiles will already be the v0.1 of 'Weird Links'.

---

### All you need is links 

https://subconscious.substack.com/p/all-you-need-is-links

I think this article can serve as a core design tenet for Links in Weird. It's filled with guiding principles that align perfectly with the core 'game loop' of our social-links system.

> ### You don’t need stars, hearts, upvotes, downvotes, you just need links
> 
> Social media, search, spam, recommendations — when faced with an abundance of information, we often find ourselves needing to separate wheat from the chaff. Many systems reach for stars, hearts, upvotes, and downvotes as quick fixes for user-generated quality signals. But what if we just used links?
> 
> ![image](https://github.com/raffomania/linkblocks/assets/583842/33664eda-feba-4ad6-a87a-4a3193108f48)
> 
> Inbound links can be used as a signal of quality. People are more likely to link to things that matter, and less likely to link to things that don’t. Sum up the backlinks to a page and you have a quality signal. This is the key insight behind [Google Pagerank](https://en.wikipedia.org/wiki/PageRank).

> ### You don’t need comments, you just need links
> 
> Comments are one of the core interaction primitives of today’s web. I can comment on your Facebook post, your Google Doc, your blog.
> 
> What is a comment? What is its structure? What is its function? It’s a bit of content that is conceptually related to a parent post. We could say that the comment points to the post, or something in the post. That’s a directional relationship.
> 
> Links also point to something. They describe a directional relationship. It’s not much of a leap to consider an inbound link a comment on the thing it links to. If we implement some form of [transclusion](https://en.wikipedia.org/wiki/Transclusion), we can express comments in terms of links. In fact, WordPress already does this with [Pingbacks](https://wordpress.org/support/article/trackbacks-and-pingbacks/).

[Curated collections](https://github.com/raffomania/linkblocks/discussions/37) for example is nothing more than a *list of links*.

To be clear though, the add-ons mentioned above (likes, comments etc.) can definitely be useful and should probably find their way into Weird. However, they should all be solved for by pure linking first so as to push the link-graph to its limits.

https://sheep.horse/2024/4/save_the_web_by_being_nice.html

_sharing links is **nice**_ 💖 

- - -

#### Importers

- We will support channel-links import from Discord, like this: https://github.com/raffomania/linkblocks/pull/61
- GitHub Awesome lists
- Bookmark Managers

#### Misc
- https://simonwillison.net/2024/Dec/22/link-blog/
- https://simonwillison.net/2024/Jul/13/give-people-something-to-link-to/
- https://news.ycombinator.com/item?id=40954879
- https://www.w3.org/Provider/Style/URI
- https://news.ycombinator.com/item?id=23865484
- https://blog.jim-nielsen.com/2024/the-subversive-hyperlink/
- https://thehistoryoftheweb.com/cool-urls-mean-something/
- https://tedium.co/2024/07/23/linkedin-complex-linking-schemes/
- https://blog.jim-nielsen.com/2023/examples-of-great-urls/

**Comment by erlend-sh** [comment]
# Example Link Lists

_Kindly add your suggestion in a reply below!_ 👇

**Reply by erlend-sh** [reply]
### Open Source Curriculum

https://github.com/erlend-sh/opensource-curriculum

**Reply by erlend-sh** [reply]
### Best-of Getting Real

- https://basecamp.com/gettingreal/02.4-fix-time-and-budget-flex-scope
- https://basecamp.com/gettingreal/03.4-embrace-constraints
- https://basecamp.com/gettingreal/05.2-it-just-doesnt-matter
- https://basecamp.com/gettingreal/06.1-race-to-running-software

**Reply by erlend-sh** [reply]
### Open WebTech Association

https://github.com/open-webtech by @secondtruth

**Reply by erlend-sh** [reply]
### Delightful Fediverse Apps

https://codeberg.org/fediverse/delightful-fediverse-apps

**Reply by zicklag** [reply]
### Privacy & Agency

- https://github.com/pluja/awesome-privacy

**Reply by zicklag** [reply]
### Databases

- https://materializedview.io/

**Reply by zicklag** [reply]
### Decentralized

- https://gitlab.com/bluesky-community1/decentralized-ecosystem/-/blob/master/README.md
- https://github.com/redecentralize/alternative-internet
- https://github.com/gdamdam/awesome-decentralized-web
- https://github.com/kgryte/awesome-peer-to-peer

**Reply by erlend-sh** [reply]
### Fedi Directory

https://fedi.directory/

**Reply by erlend-sh** [reply]
### Internet used to be fun

https://projects.kwon.nyc/internet-is-fun/

**Reply by erlend-sh** [reply]
Linkpost: https://www.rafa.ee/articles/activism-lowtech-transhumanism-sprouting-solarpunk-camp-2024/

**Reply by secondtruth** [reply]
### Fediverse Development
- https://fedi.foundation
- https://fedidevs.org

**Reply by secondtruth** [reply]
### Indie/Small Web
- https://indieweb.org
- https://small-tech.org
- https://ar.al/2020/08/07/what-is-the-small-web/

**Reply by secondtruth** [reply]
### Manifestos
- https://humanewebmanifesto.com
- https://web0.small-web.org
- https://nureinblog.at/2747-open-blogging-manifest/
- https://jeffhuang.com/designed_to_last/

**Reply by secondtruth** [reply]
### A better web for users and developers
- https://verses.xyz
- https://webwewant.fyi

**Reply by secondtruth** [reply]
### Social Coding
- https://socialcoding.org

**Reply by secondtruth** [reply]
### Essays / Articles
- https://aramzs.xyz/essays/the-internet-is-a-series-of-webs/
- https://open-webtech.github.io/owta-publications/meta/why-federation.html

**Reply by erlend-sh** [reply]
### Coops

- https://tech-coops.xyz/
- https://directory.platform.coop/

**Reply by erlend-sh** [reply]
https://github.com/yaelwrites/Big-Ass-Data-Broker-Opt-Out-List

**Reply by erlend-sh** [reply]
Topic: Horizontal scaling
- https://github.com/polyphony-chat/symfonia/issues/28

**Reply by erlend-sh** [reply]
Make a web page:
- https://htmlforpeople.com/
- https://solarpunk.cool/zines/web-zine-01/

**Reply by erlend-sh** [reply]
https://sdk.blue/

**Comment by erlend-sh** [comment]
### Ranking Lists

This https://github.com/commune-os/weird/issues/28 should have an accompanying Ranking List. The page would be titled ‘Alt-web login priorities’

The list would have two sections (i.e. headings, another concept we’ve yet to specify).

First section will be the apps already listed as TODOs in #28, with the same heading (or maybe we rename both to Planned).

Second section will be titled ‘Tentatives’. This is an exhaustive list of every small/open/indie app that could be supporting Weird for logins.

While the first list is static (editable by managers only), the second list can be reordered at will. However, as will be clearly marked, only the top app-links you’ve placed in the top 3 count as *votes* for which apps you think should support Weird-login as soon as possible. Any additional ordering you do is purely for your own clarification.

1st place is 3 votes
2nd place is 2 votes
3rd place is 1 vote.

With 20+ rank-lists submitted, the main page would have a top 10 of highest priority integrations, ranked by most vote-points.

### Page branches

I don’t know if this is the best way to put it, but the individual rankings that happen here are each a sort of branch from the main page. The main page is essentially running a function of ‘display a mass-merged aggregate of all my branches’.

**Comment by erlend-sh** [comment]
### Links graph

Every link that is added to any page (profile or project) in Weird gets added to the network-wide ‘links graph’. A web-public links aggregator would live at e.g. `weird.one/links`

Prior art:

https://github.com/sissbruecker/linkding
https://github.com/raffomania/linkblocks
https://github.com/raffomania/linkblocks/discussions/44

The links index would note stats for each link like:

- number of backlinks 
- number of occurrence in people/project’s lists
- …

Open questions:

- Is every link also its own linkspage? That seems like the easiest way to also
